From f1f06dbf1b3e9580b732c4e06b9685ea5681b380 Mon Sep 17 00:00:00 2001
From: Wentao Zhang <wentao.zhang@windriver.com>
Date: Tue, 15 Aug 2023 11:18:36 +0800
Subject: [PATCH 3/6] replace python with python3 in the test

"runtime:call" in ptest gets the following FAILED:
python: No such file or directory
replace python with python3 in the test scripts.

$export BPFTRACE_RUNTIME_TEST_EXECUTABLE=/usr/bin
$cd /usr/lib/bpftrace/ptest/tests
$python3 runtime/engine/main.py --filter="call.*"
***
[ RUN      ] call.strftime_microsecond_extension_rollover
[  FAILED  ] call.strftime_microsecond_extension_rollover
	Command: /usr/bin/bpftrace -e 'BEGIN { printf("%s - %s\n", strftime
    ("1%f", 1000000123000), strftime("1%f", 0)); exit(); }' | tail -n
    +2 | xargs -I{} python -c "print({})"
	Unclean exit code: 127
	Output: __BPFTRACE_NOTIFY_PROBES_ATTACHED\nxargs: python: No such
    file or directory\n
***

Signed-off-by: Wentao Zhang <wentao.zhang@windriver.com>
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 tests/runtime/call | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/runtime/call b/tests/runtime/call
index 2d7d9fcd..c6dcdf12 100644
--- a/tests/runtime/call
+++ b/tests/runtime/call
@@ -378,13 +378,13 @@ TIMEOUT 5
 #
 # Note we add a `1` before the timestamp b/c leading zeros (eg `0123`) is invalid integer in python.
 NAME strftime_microsecond_extension
-RUN {{BPFTRACE}} -e 'BEGIN { printf("%s - %s\n", strftime("1%f", 1000123000), strftime("1%f", 0)); exit(); }' | tail -n +2 | xargs -I{} python -c "print({})"
+RUN {{BPFTRACE}} -e 'BEGIN { printf("%s - %s\n", strftime("1%f", 1000123000), strftime("1%f", 0)); exit(); }' | tail -n +2 | xargs -I{} python3 -c "print({})"
 EXPECT 123
 TIMEOUT 1
 
 # Similar to above test but test that rolling over past 1s works as expected
 NAME strftime_microsecond_extension_rollover
-RUN {{BPFTRACE}} -e 'BEGIN { printf("%s - %s\n", strftime("1%f", 1000000123000), strftime("1%f", 0)); exit(); }' | tail -n +2 | xargs -I{} python -c "print({})"
+RUN {{BPFTRACE}} -e 'BEGIN { printf("%s - %s\n", strftime("1%f", 1000000123000), strftime("1%f", 0)); exit(); }' | tail -n +2 | xargs -I{} python3 -c "print({})"
 EXPECT 123
 TIMEOUT 1
 
-- 
2.44.0

